File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = [0] * 26
        for i in s:
            count[ord(i) - ord('a')]+=1
        for i in t:
            count[ord(i) - ord('a')]-=1
        for c in count:
            if c!=0:
                return False
        return True
        

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // sort both and check
        HashMap<Character, Integer> frequencyCounter = new HashMap<Character, Integer>();
        if ( s.length() != t.length() ) return false;
        // same length, same frequency
        for ( char c : s.toCharArray()) {
            frequencyCounter.put( c, frequencyCounter.getOrDefault(c,0) + 1);
        }
        for ( char c : t.toCharArray()) {
            frequencyCounter.put( c, frequencyCounter.getOrDefault(c,0) - 1);
        }
        for ( int frequency : frequencyCounter.values() ) {
             if ( frequency != 0) return false;
        }

        return true;

        
    }
}